x-pack/test/alerting_api_integration/security_and_spaces/group2/tests/actions/connector_types/crowdstrike.ts

```diff
@@ -0,0 +1,233 @@
+        subAction: string;
+        subActionParams: Record<string, unknown>;
+        expectedHttpCode?: number;
+        username?: string;
+        password?: string;
+        errorLogger?: (err: any) => void;
+      }) => {
+        return supertestWithoutAuth
+          .post(`${getUrlPrefix('default')}/api/actions/connector/${connectorId}/_execute`)
+          .set('kbn-xsrf', 'foo')
+          .on('error', errorLogger)
+          .auth(username, password)
+          .send({
+            params: {
+              subAction,
+              subActionParams,
+            },
+          })
+          .expect(expectedHttpCode);
+      };
+
+      describe('and user has NO privileges', () => {
+        let user: CreatedUser;
+
+        before(async () => {
+          user = await createUser({
+            username: 'read_access_user',
+            kibanaFeatures: { actions: ['read'] },
+          });
+        });
+
+        after(async () => {
+          if (user) {
+            await user.deleteUser();
+          }
+        });
+
+        for (const crowdstrikeSubAction of crowdstrikeSubActions) {
+          it(`should deny execute of ${crowdstrikeSubAction}`, async () => {
+            const execRes = await executeSubAction({
+              supertest: supertestWithoutAuth,
+              subAction: crowdstrikeSubAction,
+              subActionParams: {},
+              username: user.username,
+              password: user.password,
+              expectedHttpCode: 403,
+              errorLogger: logErrorDetails.ignoreCodes([403]),
+            });
+
+            expect(execRes.body).to.eql({
+              statusCode: 403,
+              error: 'Forbidden',
+              message: 'Unauthorized to execute a ".crowdstrike" action',
+            });
+          });
+        }
+      });
+
+      describe('and user has proper privileges', () => {
+        let user: CreatedUser;
+
+        before(async () => {
+          user = await createUser({
+            username: 'all_access_user',
+          });
+        });
+
+        after(async () => {
+          if (user) {
+            await user.deleteUser();
+            // @ts-expect-error
+            user = undefined;
+          }
+        });
+
+        for (const crowdstrikeSubAction of crowdstrikeSubActions) {
+          it(`should allow execute of ${crowdstrikeSubAction}`, async () => {
+            const {
+              // eslint-disable-next-line @typescript-eslint/naming-convention
+              body: { status, message, connector_id },
+            } = await executeSubAction({
+              supertest: supertestWithoutAuth,
+              subAction: crowdstrikeSubAction,
+              subActionParams: {},
+              username: user.username,
+              password: user.password,
+            });
+
+            expect({ status, message, connector_id }).to.eql({
+              status: 'error',
+              message: 'an error occurred while running the action',
```

Comment:
are there plans to add some sort of server simulator to simulate responses for these subactions?

Reply:
Absolutely, as a next step there's another PR that will mock responses in Cypress. However we're investigating a way to have a generic fake server for all 3rd party bidirectional vendors. It might take a moment to get it done though.